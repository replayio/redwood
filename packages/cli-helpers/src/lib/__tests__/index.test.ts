import { prettify } from '../index'

jest.mock('../paths', () => {
  return {
    getPaths: () => {
      return {
        base: '../../../../__fixtures__/example-todo-main',
      }
    },
  }
})

test('prettify formats tsx content', () => {
  const content = `import React from 'react'

  interface Props { foo: number, bar: number }

  const FooBarComponent: React.FC<Props> = ({ foo, bar }) => {
    if (foo % 3 === 0 && bar % 5 === 0) {
      return <>FooBar</>
    }

    if (foo % 3 === 0 || bar % 3 === 0) {
      return <>Foo</>;
    }

    if (foo % 5 === 0 || bar % 5 === 0) { return <>Bar</>}

    return <>{foo}, {bar}</>}`

  expect(prettify('FooBarComponent.template.tsx', content)).toMatchSnapshot()
})
