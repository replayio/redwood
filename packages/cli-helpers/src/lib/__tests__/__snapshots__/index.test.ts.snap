// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettify formats tsx content 1`] = `
"import React from 'react'

interface Props {
  foo: number
  bar: number
}

const FooBarComponent: React.FC<Props> = ({ foo, bar }) => {
  if (foo % 3 === 0 && bar % 5 === 0) {
    return <>FooBar</>
  }

  if (foo % 3 === 0 || bar % 3 === 0) {
    return <>Foo</>
  }

  if (foo % 5 === 0 || bar % 5 === 0) {
    return <>Bar</>
  }

  return (
    <>
      {foo}, {bar}
    </>
  )
}
"
`;
